starknet_py/net/rpc_schemas/rpc_schemas.py

```diff
@@ -225,26 +228,33 @@ def preprocess(self, data, **kwargs):
 
     @post_load
     def make_dataclass(self, data, **kwargs) -> BlockStateUpdate:
-        storage_diffs, contract_diffs = BlockStateUpdateSchema._extract_diffs(
-            data["state_diff"]
-        )
+        deployed_contracts = data["state_diff"]["deployed_contracts"]
+        deployed_contracts = [
+            DeployedContractSchema().load(contract) for contract in deployed_contracts
+        ]
+
+        storage_diffs = []
+        for diff in data["state_diff"]["storage_diffs"]:
+            storage_diff = StorageDiffSchema().load(
+                {"address": diff["address"], "key": diff["key"], "value": diff["value"]}
+            )
```

Comment:
Why do we need to construct this dict, couldn't we pass data directly to the schema?